{
  "uid" : "7b6702b7828ede85",
  "name" : "Basic order",
  "fullName" : "tests.test_case1#test_case1",
  "historyId" : "01ccce76c65e8c115f693828e808eb28",
  "time" : {
    "start" : 1599214122124,
    "stop" : 1599214152376,
    "duration" : 30252
  },
  "status" : "broken",
  "statusMessage" : "Exception: An exception of type TimeoutException occurred. With Element -: side_dish",
  "statusTrace" : "self = <framework.ui.menu_page.MenuPage object at 0x10f16d0d0>, loc = 'side_dish'\n\n    def __getattr__(self, loc):\n    \n        if loc in self.locators.keys():\n            self.locators[loc] = list(self.locators[loc])\n            self.locators[loc][0] = self.TYPE_OF_LOCATORS[self.locators[loc][0].lower()]\n            self.locators[loc] = tuple(self.locators[loc])\n            try:\n>               element = WebDriverWait(self.driver, self.timeout).until(\n                    EC.presence_of_element_located(self.locators[loc])\n                )\n\nvenv/lib/python3.8/site-packages/seleniumpagefactory/Pagefactory.py:41: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"9589d806-9ec0-b44c-9cb6-6973620d1cf7\")>, method = <selenium.webdriver.support.expected_conditions.presence_of_element_located object at 0x110a377c0>, message = ''\n\n    def until(self, method, message=''):\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value is not False.\"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.time() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, 'screen', None)\n                stacktrace = getattr(exc, 'stacktrace', None)\n            time.sleep(self._poll)\n            if time.time() > end_time:\n                break\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message:\n\nvenv/lib/python3.8/site-packages/selenium/webdriver/support/wait.py:80: TimeoutException\n\nDuring handling of the above exception, another exception occurred:\n\ndriver = <selenium.webdriver.firefox.webdriver.WebDriver (session=\"9589d806-9ec0-b44c-9cb6-6973620d1cf7\")>\n\n    @allure.title('Basic order')\n    def test_case1(driver):\n        with allure.step('Open homepage and setup address'):\n            driver.get('https://www.thuisbezorgd.nl/en/')\n            home_page = HomePage(driver)\n            home_page.address_field.click()\n            home_page.address_field.clear_text()\n            home_page.address_field.send_keys('8888')\n            home_page.wait_until_present(home_page.post_code)\n            home_page.post_code.click()\n            home_page.wait_until_present(home_page.post_code_alpha)\n            home_page.post_code_alpha.click()\n            home_page.wait_until_url(\n                \"https://www.thuisbezorgd.nl/en/order-takeaway-8888-alpha?search\")\n    \n        with allure.step('Choose a restaurant'):\n            restr_page = RestaurantsPage(driver)\n            restr_page.wait_until_text_in_element((By.XPATH, \"//*[@class='topbar__title-container']/button\"),'8888-alpha')\n            assert restr_page.top_bar_post_code.text == '8888-alpha'\n            restr_page.search_restaurants_field.set_text('TEST Restaurant Selenium')\n            restr_page.wait_until_present(restr_page.first_restaurant)\n            restr_page.first_restaurant.click()\n        with allure.step('Choose a dish from the menu'):\n            menu_page = MenuPage(driver)\n            menu_page.first_menu_option.click()\n>           if menu_page.side_dish.is_displayed():\n\ntests/test_case1.py:38: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <framework.ui.menu_page.MenuPage object at 0x10f16d0d0>, loc = 'side_dish'\n\n    def __getattr__(self, loc):\n    \n        if loc in self.locators.keys():\n            self.locators[loc] = list(self.locators[loc])\n            self.locators[loc][0] = self.TYPE_OF_LOCATORS[self.locators[loc][0].lower()]\n            self.locators[loc] = tuple(self.locators[loc])\n            try:\n                element = WebDriverWait(self.driver, self.timeout).until(\n                    EC.presence_of_element_located(self.locators[loc])\n                )\n            except (StaleElementReferenceException, NoSuchElementException, TimeoutException) as e:\n>               raise Exception(\"An exception of type \" + type(e).__name__ + \" occurred. With Element -: \" + loc)\nE               Exception: An exception of type TimeoutException occurred. With Element -: side_dish\n\nvenv/lib/python3.8/site-packages/seleniumpagefactory/Pagefactory.py:45: Exception",
  "flaky" : false,
  "newFailed" : false,
  "beforeStages" : [ {
    "name" : "driver",
    "time" : {
      "start" : 1599214116226,
      "stop" : 1599214122121,
      "duration" : 5895
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  } ],
  "testStage" : {
    "status" : "broken",
    "statusMessage" : "Exception: An exception of type TimeoutException occurred. With Element -: side_dish",
    "statusTrace" : "self = <framework.ui.menu_page.MenuPage object at 0x10f16d0d0>, loc = 'side_dish'\n\n    def __getattr__(self, loc):\n    \n        if loc in self.locators.keys():\n            self.locators[loc] = list(self.locators[loc])\n            self.locators[loc][0] = self.TYPE_OF_LOCATORS[self.locators[loc][0].lower()]\n            self.locators[loc] = tuple(self.locators[loc])\n            try:\n>               element = WebDriverWait(self.driver, self.timeout).until(\n                    EC.presence_of_element_located(self.locators[loc])\n                )\n\nvenv/lib/python3.8/site-packages/seleniumpagefactory/Pagefactory.py:41: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"9589d806-9ec0-b44c-9cb6-6973620d1cf7\")>, method = <selenium.webdriver.support.expected_conditions.presence_of_element_located object at 0x110a377c0>, message = ''\n\n    def until(self, method, message=''):\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value is not False.\"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.time() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, 'screen', None)\n                stacktrace = getattr(exc, 'stacktrace', None)\n            time.sleep(self._poll)\n            if time.time() > end_time:\n                break\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message:\n\nvenv/lib/python3.8/site-packages/selenium/webdriver/support/wait.py:80: TimeoutException\n\nDuring handling of the above exception, another exception occurred:\n\ndriver = <selenium.webdriver.firefox.webdriver.WebDriver (session=\"9589d806-9ec0-b44c-9cb6-6973620d1cf7\")>\n\n    @allure.title('Basic order')\n    def test_case1(driver):\n        with allure.step('Open homepage and setup address'):\n            driver.get('https://www.thuisbezorgd.nl/en/')\n            home_page = HomePage(driver)\n            home_page.address_field.click()\n            home_page.address_field.clear_text()\n            home_page.address_field.send_keys('8888')\n            home_page.wait_until_present(home_page.post_code)\n            home_page.post_code.click()\n            home_page.wait_until_present(home_page.post_code_alpha)\n            home_page.post_code_alpha.click()\n            home_page.wait_until_url(\n                \"https://www.thuisbezorgd.nl/en/order-takeaway-8888-alpha?search\")\n    \n        with allure.step('Choose a restaurant'):\n            restr_page = RestaurantsPage(driver)\n            restr_page.wait_until_text_in_element((By.XPATH, \"//*[@class='topbar__title-container']/button\"),'8888-alpha')\n            assert restr_page.top_bar_post_code.text == '8888-alpha'\n            restr_page.search_restaurants_field.set_text('TEST Restaurant Selenium')\n            restr_page.wait_until_present(restr_page.first_restaurant)\n            restr_page.first_restaurant.click()\n        with allure.step('Choose a dish from the menu'):\n            menu_page = MenuPage(driver)\n            menu_page.first_menu_option.click()\n>           if menu_page.side_dish.is_displayed():\n\ntests/test_case1.py:38: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <framework.ui.menu_page.MenuPage object at 0x10f16d0d0>, loc = 'side_dish'\n\n    def __getattr__(self, loc):\n    \n        if loc in self.locators.keys():\n            self.locators[loc] = list(self.locators[loc])\n            self.locators[loc][0] = self.TYPE_OF_LOCATORS[self.locators[loc][0].lower()]\n            self.locators[loc] = tuple(self.locators[loc])\n            try:\n                element = WebDriverWait(self.driver, self.timeout).until(\n                    EC.presence_of_element_located(self.locators[loc])\n                )\n            except (StaleElementReferenceException, NoSuchElementException, TimeoutException) as e:\n>               raise Exception(\"An exception of type \" + type(e).__name__ + \" occurred. With Element -: \" + loc)\nE               Exception: An exception of type TimeoutException occurred. With Element -: side_dish\n\nvenv/lib/python3.8/site-packages/seleniumpagefactory/Pagefactory.py:45: Exception",
    "steps" : [ {
      "name" : "Open homepage and setup address",
      "time" : {
        "start" : 1599214122124,
        "stop" : 1599214139975,
        "duration" : 17851
      },
      "status" : "passed",
      "steps" : [ ],
      "attachments" : [ ],
      "parameters" : [ ],
      "stepsCount" : 0,
      "attachmentsCount" : 0,
      "shouldDisplayMessage" : false,
      "hasContent" : false
    }, {
      "name" : "Choose a restaurant",
      "time" : {
        "start" : 1599214139975,
        "stop" : 1599214141915,
        "duration" : 1940
      },
      "status" : "passed",
      "steps" : [ ],
      "attachments" : [ ],
      "parameters" : [ ],
      "stepsCount" : 0,
      "attachmentsCount" : 0,
      "shouldDisplayMessage" : false,
      "hasContent" : false
    }, {
      "name" : "Choose a dish from the menu",
      "time" : {
        "start" : 1599214141915,
        "stop" : 1599214152372,
        "duration" : 10457
      },
      "status" : "broken",
      "statusMessage" : "Exception: An exception of type TimeoutException occurred. With Element -: side_dish\n",
      "statusTrace" : "  File \"/Users/aigulgabi/PycharmProjects/takeAwayAssignment/tests/test_case1.py\", line 38, in test_case1\n    if menu_page.side_dish.is_displayed():\n  File \"/Users/aigulgabi/PycharmProjects/takeAwayAssignment/venv/lib/python3.8/site-packages/seleniumpagefactory/Pagefactory.py\", line 45, in __getattr__\n    raise Exception(\"An exception of type \" + type(e).__name__ + \" occurred. With Element -: \" + loc)\n",
      "steps" : [ ],
      "attachments" : [ ],
      "parameters" : [ ],
      "stepsCount" : 0,
      "attachmentsCount" : 0,
      "shouldDisplayMessage" : true,
      "hasContent" : true
    } ],
    "attachments" : [ {
      "uid" : "b69b97ba0394b279",
      "name" : "log",
      "source" : "b69b97ba0394b279.txt",
      "type" : "text/plain",
      "size" : 374
    }, {
      "uid" : "83336ac3719b49ca",
      "name" : "stderr",
      "source" : "83336ac3719b49ca.txt",
      "type" : "text/plain",
      "size" : 294
    } ],
    "parameters" : [ ],
    "stepsCount" : 3,
    "attachmentsCount" : 2,
    "shouldDisplayMessage" : true,
    "hasContent" : true
  },
  "afterStages" : [ {
    "name" : "driver::1",
    "time" : {
      "start" : 1599214152474,
      "stop" : 1599214152477,
      "duration" : 3
    },
    "status" : "broken",
    "statusMessage" : "AttributeError: 'Function' object has no attribute 'rep_call'\n",
    "statusTrace" : "  File \"/Users/aigulgabi/PycharmProjects/takeAwayAssignment/venv/lib/python3.8/site-packages/allure_commons/_allure.py\", line 185, in __call__\n    return self._fixture_function(*args, **kwargs)\n  File \"/Users/aigulgabi/PycharmProjects/takeAwayAssignment/venv/lib/python3.8/site-packages/_pytest/fixtures.py\", line 795, in _teardown_yield_fixture\n    next(it)\n  File \"/Users/aigulgabi/PycharmProjects/takeAwayAssignment/tests/conftest.py\", line 56, in driver\n    if request.node.rep_call.failed:\n",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : true,
    "hasContent" : true
  }, {
    "name" : "driver::tear_down",
    "time" : {
      "start" : 1599214152478,
      "stop" : 1599214152872,
      "duration" : 394
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  } ],
  "labels" : [ {
    "name" : "parentSuite",
    "value" : "tests"
  }, {
    "name" : "suite",
    "value" : "test_case1"
  }, {
    "name" : "host",
    "value" : "MacBook-Pro-AigulGabi.local"
  }, {
    "name" : "thread",
    "value" : "98489-MainThread"
  }, {
    "name" : "framework",
    "value" : "pytest"
  }, {
    "name" : "language",
    "value" : "cpython3"
  }, {
    "name" : "package",
    "value" : "tests.test_case1"
  }, {
    "name" : "resultFormat",
    "value" : "allure2"
  } ],
  "parameters" : [ ],
  "links" : [ ],
  "hidden" : false,
  "retry" : false,
  "extra" : {
    "severity" : "normal",
    "retries" : [ {
      "uid" : "2ce1aa463eb1d3e8",
      "status" : "passed",
      "time" : {
        "start" : 1599170218760,
        "stop" : 1599170230686,
        "duration" : 11926
      }
    }, {
      "uid" : "3765171f7b63bca",
      "status" : "passed",
      "time" : {
        "start" : 1599170180479,
        "stop" : 1599170190844,
        "duration" : 10365
      }
    }, {
      "uid" : "2a282fe545e7dc47",
      "status" : "broken",
      "statusDetails" : "selenium.common.exceptions.ElementNotInteractableException: Message: Element <span class=\"button_add_value\"> could not be scrolled into view",
      "time" : {
        "start" : 1599168795058,
        "stop" : 1599168800334,
        "duration" : 5276
      }
    }, {
      "uid" : "d2da3f27fbae30e8",
      "status" : "broken",
      "statusDetails" : "selenium.common.exceptions.ElementNotInteractableException: Message: Element <span class=\"button_add_value\"> could not be scrolled into view",
      "time" : {
        "start" : 1599168619359,
        "stop" : 1599168633967,
        "duration" : 14608
      }
    }, {
      "uid" : "6a8e28967af8d3d5",
      "status" : "broken",
      "statusDetails" : "Exception: An exception of type TimeoutException occurred. With Element -: address",
      "time" : {
        "start" : 1599168211659,
        "stop" : 1599168227448,
        "duration" : 15789
      }
    }, {
      "uid" : "f1828c87db9bd023",
      "status" : "broken",
      "statusDetails" : "Exception: An exception of type TimeoutException occurred. With Element -: address",
      "time" : {
        "start" : 1599168061102,
        "stop" : 1599168076417,
        "duration" : 15315
      }
    }, {
      "uid" : "301ce1616128be36",
      "status" : "broken",
      "statusDetails" : "Exception: An exception of type TimeoutException occurred. With Element -: address",
      "time" : {
        "start" : 1599167976636,
        "stop" : 1599167991894,
        "duration" : 15258
      }
    }, {
      "uid" : "6ccd28eed3b78c0f",
      "status" : "broken",
      "statusDetails" : "Exception: An exception of type TimeoutException occurred. With Element -: first_restaurant",
      "time" : {
        "start" : 1599167834607,
        "stop" : 1599167850594,
        "duration" : 15987
      }
    }, {
      "uid" : "37b8cda1ab1cc9f0",
      "status" : "broken",
      "statusDetails" : "Exception: An exception of type TimeoutException occurred. With Element -: first_restaurant",
      "time" : {
        "start" : 1599167781592,
        "stop" : 1599167795780,
        "duration" : 14188
      }
    }, {
      "uid" : "33ca87c59b5b4653",
      "status" : "broken",
      "statusDetails" : "Exception: An exception of type TimeoutException occurred. With Element -: first_restaurant",
      "time" : {
        "start" : 1599167708979,
        "stop" : 1599167724573,
        "duration" : 15594
      }
    }, {
      "uid" : "3fc8911e254e11db",
      "status" : "passed",
      "time" : {
        "start" : 1599148479887,
        "stop" : 1599148491911,
        "duration" : 12024
      }
    }, {
      "uid" : "919303f6767cd271",
      "status" : "passed",
      "time" : {
        "start" : 1599148082359,
        "stop" : 1599148094028,
        "duration" : 11669
      }
    }, {
      "uid" : "5a8704afa1d60737",
      "status" : "broken",
      "statusDetails" : "Exception: An exception of type TimeoutException occurred. With Element -: address",
      "time" : {
        "start" : 1599147898668,
        "stop" : 1599147914085,
        "duration" : 15417
      }
    }, {
      "uid" : "4a967f015d29e4d",
      "status" : "passed",
      "time" : {
        "start" : 1599147716160,
        "stop" : 1599147733445,
        "duration" : 17285
      }
    }, {
      "uid" : "e8a94d4914c213e3",
      "status" : "broken",
      "statusDetails" : "Exception: An exception of type TimeoutException occurred. With Element -: address",
      "time" : {
        "start" : 1599147338032,
        "stop" : 1599147358752,
        "duration" : 20720
      }
    }, {
      "uid" : "b1d5f09c8e0863de",
      "status" : "broken",
      "statusDetails" : "Exception: An exception of type TimeoutException occurred. With Element -: first_restaurant",
      "time" : {
        "start" : 1599080426157,
        "stop" : 1599080440578,
        "duration" : 14421
      }
    }, {
      "uid" : "2539aea443906de0",
      "status" : "passed",
      "time" : {
        "start" : 1599080193323,
        "stop" : 1599080204135,
        "duration" : 10812
      }
    }, {
      "uid" : "8c4dab642a28ae33",
      "status" : "passed",
      "time" : {
        "start" : 1599079783435,
        "stop" : 1599079795422,
        "duration" : 11987
      }
    } ],
    "categories" : [ {
      "name" : "Test defects",
      "matchedStatuses" : [ ],
      "flaky" : false
    } ],
    "tags" : [ ]
  },
  "source" : "7b6702b7828ede85.json",
  "parameterValues" : [ ]
}